<launch>
  <!-- <arg name="sensor_name" default="D455" /> -->

  <!-- <arg name="exp_distance" default="3000.0" /> -->

  <!-- <arg name="record_p" default="0" /> -->

  <arg name="zkey" default="3200.0" />
  <arg name="min_gap_of_occluding_boundary" default="50.0" />
  <arg name="min_pixels_as_object" default="200" />
  <arg name="min_overlap_rate" default="0.8" />
  
  <arg name="use_mask_p" default="0" />
  <!-- <arg name="use_region_p" default="0" /> -->
  <arg name="display_images_p" default="1" />
  <arg name="publish_images_p" default="1" />
  <arg name="publish_markers_p" default="0" />

  <arg name="save_images_p" default="0" />
  <arg name="dirname_log" default="$(env PWD)/Data/" />

  <arg name="filename_mask" default="MASK.png" />
  <!-- <arg name="filename_region" default="Reg.png" /> -->
  <arg name="intopic" default="/sensing_unit/depth_registered/points_throttle" />
  
  <group ns="processing_unit">
    <remap from="/camera/depth_registered/points"
           to="$(arg intopic)" />

    <node name="measurer" pkg="emulated_srs" type="obstacle_measurer"
	  output="screen" cwd="node">
      
      <!-- <param name="sensor_name" value="$(arg sensor_name)" /> -->

      <!-- <param name="dist_testpiece" value="$(arg exp_distance)" /> -->

      <param name="use_mask_p" value="$(arg use_mask_p)" />
      <!-- <param name="use_region_p" value="$(arg use_region_p)" /> -->
      
      <param name="zkey" value="$(arg zkey)" />
      <param name="min_gap_of_occluding_boundary"
	     value="$(arg min_gap_of_occluding_boundary)" />
      <param name="min_pixels_as_object" value="$(arg min_pixels_as_object)" />
      <param name="min_overlap_rate" value="$(arg min_overlap_rate)" />

      <param name="filename_mask" value="$(arg filename_mask)" />
      <!-- <param name="filename_region" value="$(arg filename_region)" /> -->
      <param name="dirname_log" value="$(arg dirname_log)" />
      
      <param name="display_images_p" value="$(arg display_images_p)" />
      <param name="publish_images_p" value="$(arg publish_images_p)" />
      <param name="publish_markers_p" value="$(arg publish_markers_p)" />
      <param name="save_images_p" value="$(arg save_images_p)" />
    </node>

  </group>

</launch>

 
